test/e2e/windows/gmsa_full.go

```diff
@@ -163,6 +169,74 @@ var _ = SIGDescribe("[Feature:Windows] GMSA Full [Serial] [Slow]", func() {
 				return true
 			}, 1*time.Minute, 1*time.Second).Should(gomega.BeTrue())
 		})
+
+		ginkgo.It("can read and write file to remote folder", func() {
+			defer ginkgo.GinkgoRecover()
+
+			ginkgo.By("finding the worker node that fulfills this test's assumptions")
+			nodes := findPreconfiguredGmsaNodes(f.ClientSet)
+			if len(nodes) != 1 {
+				e2eskipper.Skipf("Expected to find exactly one node with the %q label, found %d", gmsaFullNodeLabel, len(nodes))
+			}
+			node := nodes[0]
+
+			ginkgo.By("retrieving the contents of the GMSACredentialSpec custom resource manifest from the node")
+			crdManifestContents := retrieveCRDManifestFileContents(f, node)
+
+			ginkgo.By("downloading the GMSA webhook deploy script")
+			deployScriptPath, err := downloadFile(gmsaWebhookDeployScriptURL)
+			defer func() { os.Remove(deployScriptPath) }()
+			if err != nil {
+				framework.Failf(err.Error())
+			}
+
+			ginkgo.By("deploying the GMSA webhook")
+			webhookCleanUp, err := deployGmsaWebhook(f, deployScriptPath)
+			defer webhookCleanUp()
+			if err != nil {
+				framework.Failf(err.Error())
+			}
+
+			ginkgo.By("creating the GMSA custom resource")
+			customResourceCleanup, err := createGmsaCustomResource(f.Namespace.Name, crdManifestContents)
+			defer customResourceCleanup()
+			if err != nil {
+				framework.Failf(err.Error())
+			}
+
+			ginkgo.By("creating an RBAC role to grant use access to that GMSA resource")
+			rbacRoleName, rbacRoleCleanup, err := createRBACRoleForGmsa(f)
+			defer rbacRoleCleanup()
+			if err != nil {
+				framework.Failf(err.Error())
+			}
+
+			ginkgo.By("creating a service account")
+			serviceAccountName := createServiceAccount(f)
+
+			ginkgo.By("binding the RBAC role to the service account")
+			bindRBACRoleToServiceAccount(f, serviceAccountName, rbacRoleName)
+
+			ginkgo.By("creating a pod using the GMSA cred spec")
+			podName := createPodWithGmsa(f, serviceAccountName)
+
+			ginkgo.By("getting the ip of GMSA domain")
+			gmsaDomainIP := getGmsaDomainIP(f, podName)
+
+			ginkgo.By("creating a pod using the GMSA cred spec")
+			filePath := "\\\\" + gmsaDomainIP + "\\" + gmsaSharedFolder + "\\write_test.txt"
+			_, _ = runKubectlExecInNamespace(f.Namespace.Name, podName, "--", "powershell.exe", "-Command", "echo 'This is a test file.' > "+filePath)
+
+			ginkgo.By("checking that file can be written to the remote folder successfully")
+			gomega.Eventually(func() bool {
```

Comment:
any reason this is in eventually and the write above isn't? Should these be separate pods to ensure we aren't doing anything funny with writing to local paths?

Reply:
Thanks for the advice! I updated the code and moved the write into the eventually function. 
Also I added a comment to the filepath variable to tell folks not to change it. "gmsaSharedFolder" variable can be changed to other folder, but as long as the "filePath" variable(line229) doesn't change in their pr, it would still be a remote folder and the test takes effect. Do you think it's good enough to prevent writing to local paths?

Reply:
yes I think this makes sense.